### DISCUSSION TITLE
Classic Stand-up Roller Coaster

### DISCUSSION THREAD

**Discussion by SpartanFrederic104** [OpenRCT2/OpenRCT2#18171] [Feature requests] [answered]
For those who have played both RCT1 and RCT2, the Stand-up Roller Coaster (then known as the Stand-up Steel Roller Coaster) was changed between RCT1 and RCT2:
- The banked curve angle is shallower in RCT1 compared to RCT2. This was also the case with the Wooden Roller Coaster.
- The RCT1 stand-up coaster does not have banked helixes, while the RCT2 rendition does. The Wooden Roller Coaster in RCT1 did not have banked helixes either.
- The RCT1 trains don't have an empty tail car at the back, whereas the RCT2 trains do.
- The reverse-incline launched shuttle and powered launch operating modes were removed from the RCT2 stand-up coaster, but they were later re-added in OpenRCT2 for parity with the RCT1 rendition.

With the RCT1 Steel Mini and Wooden Roller Coasters now in the game as the Classic Mini and Classic Wooden Roller Coasters, respectively, the RCT1 stand-up coaster currently remains the last coaster type from RCT1 that isn't available yet.

**Comment by Gymnasiast** [comment]
That is true, but is there any point to adding it, apart from ticking a box? The Classic Wooden at least harkens back to old wooden roller coasters, and they way it handles colour schemes is also different from the Wooden “Twister”, which was another reason to include it for RCT1 compat.

But there is not much special about the RCT1 Stand-Up, apart from some slight visual differences that I would personally chalk up as “better artwork”.

**Comment by Generalcamo** [comment]
We could honestly consider adding more bank options, and fully merge the two track types. Rather than an arbitrary split like this.

**Reply by Gymnasiast** [reply]
How would that work with existing vehicles? All vehicles just have sprites for a single banking degree.

**Reply by spacek531** [reply]
With the feature to disable unsupported track elements by vehicle and the changes to vehicle sprite groups, it seems like it would work relatively simply.

Add the sprite groups to the list of sprite groups, add the paint code to paint the sprite groups, add the track pieces to the list of track pieces, add the track groups to the list of track groups, and to tie everything together, tie the sprite groups to the track groups using the function.

**Reply by 733737** [reply]
xtreme is already working on adding multiple bank types to the game iirc, in addition multiple bank types would be useful for creating elements similar to parkitect's style

**Reply by ZeeMaji** [reply]
I think this solution would work quite well, we could also potentially use it for the wooden coaster (wooden twister in rct1)'s subtle banking angle difference in RCT1.

**Comment by ZeeMaji** [comment]
After thinking about it a bit more, i think this would be the best solution rather than just adding the different banking sprites to the rct2 stand-up.

As i recently discovered that the rct1 stand-up has a higher support limit than the rct2 stand-up (29 units for rct1, 23 for rct2)
While we could just add that to the rct2 stand-up i feel like it's just going to make the rct2 stand-up even more absurdly unrealistic than it already is with the other rct1 parity changes that were made to it.

Especially in regards to the extra operating modes, they're not only really absurd to have on the stand-up but also greatly impact the balance of the rct2 stand-up coaster as the powered launch modes make it much better than it really should be.

The classic stand-up would resolve that by separating all of the unrealistic rct1 parity aspects onto a different ride type ala the classic mini coaster.

**Reply by SpartanFrederic104** [reply]
That reminds me, I actually added those operating modes to the RCT2 stand-up several years ago. Should those be removed when the RCT1 stand-up gets added?

**Reply by ZeeMaji** [reply]
Definitely.

**Reply by Gymnasiast** [reply]
Hm, keeping the RCT2 counterpart clean (and remove unrealistic parity changes) does seem like a good reason to me. I do think the RCT1 vehicle (and by extension, the ride type itself) should be marked as a compatibility object, though. (You can already do that even before the necessary bits have been merged, the field will just get ignored until that point.)

**Reply by ZeeMaji** [reply]
I think it would be fine to leave it as a normal object, it could be interesting to use in future custom scenarios as the differences make it pretty separate from the rct2 stand-up balance wise.

**Reply by SpartanFrederic104** [reply]
So I recently thought of the RCT1 stand-up having this description in the build menu:
_"An intense, older-style steel looping roller coaster where the riders ride in a standing position"_
Would that work?

Also on the topic of keeping RCT2 counterparts of RCT1 coaster types clean, doesn't the Junior Roller Coaster (the RCT2 version of the Steel Mini Roller Coaster without steep drops) still have the reverse-incline launched shuttle mode?

**Reply by Gymnasiast** [reply]
No, that mode was removed back in September (#18001).

**Reply by SpartanFrederic104** [reply]
Sorry to bump this as it will almost be a full year since the last reply, but I have the following questions:
- Does the RCT1 stand-up coaster have a slight difference (if any) from the RCT2 stand-up coaster in terms of stat calculations?
- Are there any unique track color presets for the RCT1 stand-up that aren't featured on the RCT2 variant?

**Reply by Gymnasiast** [reply]
I think @ZeeMaji should be able to answer this. In the meantime though, I have committed to implementing the Classic Stand-Up roller coaster.

**Reply by ZeeMaji** [reply]
The rating calculations & color presets are the same between the rct1 stand-up & rct2 stand-up.

**Comment by ZeeMaji** [comment, marked as answer]
Implemented in #21642
